### ISSUE TITLE
Sysroot Path Incorrect Post NDK R19

### ISSUE THREAD

**Issue by GEMISIS** [facebook/buck#2700]
Per https://github.com/android/ndk/issues/1407, the sysroot path for NDK is now contained in <NDK>/toolchains/llvm/prebuilt/<host-tag>/sysroot instead of the top level directory. This means that buck is looking in the wrong spot for the sysroot directory at https://github.com/facebook/buck/blob/5444652af489563ddaf595a634341dd14fd41399/src/com/facebook/buck/android/toolchain/ndk/impl/NdkCxxPlatforms.java#L1209

**Comment by GEMISIS** [comment]
Looks like my original bug was an issue with my paths, but there is still an issue in that GCC related tools are being linked in the toolchain, which are no longer included with the NDK after R22.

**Comment by GEMISIS** [comment]
Actually, looks at this again, it may also be that it's failing before getting to what my original comment was discussing?

Per https://github.com/android/ndk/issues/1407, the sysroot path for NDK is now contained in /toolchains/llvm/prebuilt//sysroot instead of the top level directory. This means that buck is looking in the wrong spot for the sysroot directory at https://github.com/facebook/buck/blob/5444652af489563ddaf595a634341dd14fd41399/src/com/facebook/buck/android/toolchain/ndk/impl/NdkCxxPlatforms.java#L1209

It seems that NDK support may not have been kept up to date?

**Comment by GEMISIS** [comment]
Yeah, it looks like in the NdkCxxPlatforms.java file, there's a lot of updates that need to happen for more recent versions of the NDK, as everything after R19 is completely broken. The latest I've managed to get building on Windows is NDK version 19.2.5345600.

**Comment by LegNeato** [comment]
https://github.com/facebook/buck/pull/2685
